index.js

```diff
@@ -0,0 +1,67 @@
+const exec = require("child_process");
+const tmp = require("tmp");
+const fs = require("fs")
+const path = require("path");
+const { ENOENT } = require("constants");
+
+class SoxException extends Error {
+    constructor(message) {
+        super(message);
+    }
+}
+
+async function runSox(args) {
+    const sox = await exec.spawnSync(
+        "sox", args, { encoding: "utf8" }
+    );
+    if (sox.error !== undefined) {
+        console.log(sox.error.code);
+        if (sox.error.code === "ENOENT") {
+            throw new SoxException(`sox not found, install it first`);
+        }
+        throw new SoxException(`sox exited: ${sox.error}`);
+    }
+    if (sox.status != 0) {
+        throw new SoxException(`sox exited: ${sox.status} ${sox.stderr}`);
+    }
+}
+
+async function Process(src, dst) {
+    const tmpFileName1 = tmp.tmpNameSync({ postfix: ".wav" });
+    const tmpFileName2 = tmp.tmpNameSync({ postfix: ".wav" });
+    await runSox([src, "--norm", "-c 1", tmpFileName1]);
+    await runSox([tmpFileName1, tmpFileName2, "silence", "1", "0.3", "1%", "reverse"]);
+    await runSox([tmpFileName2, dst, "silence", "1", "0.1", "0.1%", "reverse", "pad", "0.1", "0.4"]);
+}
+
+
+async function main() {
+    const srcDir = process.argv[2];
+    const dstDir = process.argv[3];
+    let hasErrors = false;
+    if (srcDir == undefined || dstDir == undefined) {
+        console.log("node index.js <src_dir> <dst_dir>");
+        return;
+    }
+    var entries = await fs.promises.readdir(srcDir);
```

Comment:
Для скриптов я бы пользовался синхронными функциями.

```suggestion
    const entries = fs.readdirSync(srcDir);
```